pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go

```diff
@@ -71,6 +71,10 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
+# CONTAINER_TOOL defines the container tool to be used for building images.
```

Comment:
All makefile targets does not work with ANY other container tool
For example  `docker-buildx: test ## Build and push docker image for the manager for cross-platform support` will really work within any container tool or podman?

So, if you would like to create the env var here it is fine but I do not think that we can say that it can be used with podman . Also, we would need to highlight that it is only tested/support with docker by default. We do not test other tools within. 

Is that make sense?

Therefore, if we provide the further clarifications I am OK within.

Reply:
Hello @camilamacedo86 

Iirc podman does have a buildx subcommand, but I agree as well that not all container tools will support all Makefile targets. Where would you prefer to add verbiage on this? Where the $CONTAINER_TOOL variable is set?

Reply:
Hi @redhatrises, 

Thank you for the contribution and for share that. 
Please, feel free to wording it as you see fit.  I think anything like follows would be great. 

> # CONTAINER_TOOL defines the container tool to be used for building images. Note that the target commands are only tested with Docker which is scaffolded by default. However, you might want to replace it to use other tools. (i.e. podman)

Also, could you please only apply the change in the go/v4 plugin?
The plugin v3 is deprecated so ideally we should only perform changes on it if is some critical bug. So we avoid introduce regressions either motivate users to upgrade their projects.     

Again, thank you a lot for your contribution 🥇 


/ok-to-test

Reply:
Thoughts on taking in a build-cmd directly? Something like [this](https://sdk.operatorframework.io/docs/cli/operator-sdk_pkgman-to-bundle/#options). If the user provides an entire cmd through an envVar we run it, or we default to docker commands. This will enable customizations on path of where binary can be located too. 
cc: @camilamacedo86

Reply:
> Hi @redhatrises,
> 
> Thank you for the contribution and for share that. Please, feel free to wording it as you see fit. I think anything like follows would be great.
> 
> > CONTAINER_TOOL defines the container tool to be used for building images. Be aware that the target commands are only tested with Docker which is scaffolded by default. However, you might want to replace it to use other tools. (i.e. podman)
> 
> Also, could you please only apply the change in the go/v4 plugin? The plugin v3 is deprecated so ideally we should only perform changes on it if is some critical bug. So we avoid introduce regressions either motivate users to upgrade their projects.
> 
> Again, thank you a lot for your contribution 🥇
> 
> /ok-to-test

@camilamacedo86 I added your wording, because it works well. I also removed updates to the v3 plugin.